qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,536 @@
+    sparse_pauli = None
+    for term in fer_op_list:
+        term_type, facs = _analyze_term(_operator_string(term))
+        if facs[0][1] == "-":  # keep only one of h.c. pair
+            continue
+        ## Following only filters h.c. of some number-excitation op
+        if facs[0][0] == facs[1][0]:  # first op is number op, which is it's own h.c.
+            if len(facs) > 2 and facs[2][1] == "-":  # So, look at next op to skip h.c.
+                continue
+
+        if term_type == TermType.NUMBER:  # a^\dagger_p a_p
+            p = facs[0][0]  # pylint: disable=invalid-name
+            h1_pq = _operator_coefficient(term)
+            sparse_pauli = _add_sparse_pauli(sparse_pauli, _number_operator(edge_list, p, h1_pq))
+            continue
+
+        if term_type == TermType.EXCITATION:
+            (p, q) = [facs[i][0] for i in range(2)]  # p < q always   # pylint: disable=invalid-name
+            h1_pq = _operator_coefficient(term)
+            sparse_pauli = _add_sparse_pauli(
+                sparse_pauli, _excitation_operator(edge_list, p, q, h1_pq)
+            )
+
+        else:
+            facs_reordered, phase = _to_physicist_index_order(facs)
+            h2_pqrs = phase * _operator_coefficient(term)
+            (p, q, r, s) = [facs_reordered[i][0] for i in range(4)]  # pylint: disable=invalid-name
+            if term_type == TermType.DOUBLE_EXCITATION:
+                sparse_pauli = _add_sparse_pauli(
+                    sparse_pauli, _double_excitation(edge_list, p, q, r, s, h2_pqrs)
+                )
+            elif term_type == TermType.COULOMB_EXCHANGE:
+                sparse_pauli = _add_sparse_pauli(
+                    sparse_pauli, _coulomb_exchange(edge_list, p, q, s, h2_pqrs)
+                )
+            elif term_type == TermType.NUMBER_EXCITATION:
+                # Note that h2_pqrs is not divided by 2 here, as in the aqua code
+                sparse_pauli = _add_sparse_pauli(
+                    sparse_pauli, _number_excitation(edge_list, p, q, r, s, h2_pqrs)
+                )
+            else:
+                raise ValueError("Unknown interaction: ", term_type)
+
+    return sparse_pauli
+
+
+def _add_sparse_pauli(qubit_op1: SparsePauliOp, qubit_op2: SparsePauliOp) -> SparsePauliOp:
+    """
+    Return `qubit_op1` + `qubit_op2`, except when either one is `None`.
+    In the latter case, return the one that is not `None`. In other words, assume
+    `None` signifies the additive identity.
+    """
+    if qubit_op1 is None:
+        return qubit_op2
+    elif qubit_op2 is None:
+        return qubit_op1
+    else:
+        return qubit_op1 + qubit_op2
+
+
+def _analyze_term(term_str: str) -> Tuple[TermType, List]:
+    """
+    Return the type of interaction represented by `term_str` and
+    a list of the factors and their indices in `term_str`.
+
+    The types of interaction are NUMBER, EXCITATION, COULOMB_EXCHANGE, NUMBER_EXCITATION,
+    DOUBLE_EXCITATION.
```

Comment:
Maybe refer to `TermType` rather than duplicate the Enum values here.

Reply:
I pushed a commit that may do you are thinking of here.